### ISSUE TITLE
homework1 loss setting

### ISSUE THREAD

**Issue by PaParaZz1** [PeiqinSun/tf-tutorials#46]
I find there is something doubtful in train.py in homework1.
Here is the code in tutorials in train.py 66-70 line

```
    correct_pred = tf.equal(tf.cast(tf.argmax(preds, 1), dtype=tf.int32),
                            tf.cast(tf.argmax(label_onehot, 1), dtype=tf.int32))
    accuracy = tf.reduce_mean(tf.cast(correct_pred, tf.float32))
    loss_reg = tf.add_n(tf.get_collection(tf.GraphKeys.REGULARIZATION_LOSSES))
    loss = tf.losses.softmax_cross_entropy(label_onehot, logits) + loss_reg
```
In the homework1 requirements, we need to use different replacements of softmax, but in practical loss function code, we always use the 'softmax_cross_entropy'  to calculate between logits and label_onehot, but the variable 'logits' is the output of the network, rather than what we really changed(variable pred). We only change the way to calculate correct prediction. **Does it really make sense???**

**Comment by Somedaywilldo** [comment]
附议，如果第一题只是改个default的loss命令行参数，我觉得毫无意义

**Comment by perrylover** [comment]
为方便大家参考，这里给出softmax替代方式计算logits且可通过命令行参数控制，但每个替代方式对应的loss还是需要大家自己实现的

**Comment by PaParaZz1** [comment]
ok. Thank you for your answer.
